src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,149 @@
+                "Hello! I'm KwunTalk!\nWhat can I do for you?\n" +
+                DIVIDER;
+    }
+
+
+    /**
+     * Shows the goodbye message to the user.
+     */
+    public String goodbye() {
+        return DIVIDER + "Bye. Hope to see you again soon!\n" + DIVIDER;
+    }
+
+
+    /**
+     * Shows the lists of tasks.
+     *
+     * @param taskList List of tasks.
+     */
+    public String listTasks(TaskList taskList) {
+        if (taskList.isEmpty()) {
+            return DIVIDER + "Your list is currently empty.\n" + DIVIDER;
+
+        } else {
+            StringBuilder sb = new StringBuilder();
+
+            for (int i = 1; i <= taskList.getLength(); i++) {
+                sb.append(String.format("%d. %s\n", i, taskList.get(i)));
+            }
+            return DIVIDER + "Here are the tasks in your list:\n" + sb + DIVIDER;
+        }
+    }
+
+
+    /**
+     * Shows the message when a task is deleted from the list.
+     *
+     * @param task Task that was deleted.
+     * @param taskTotal Total number of tasks in the list after deleting.
+     */
+    public String deleteTask(Task task, int taskTotal) {
+        String s = String.format("OK. I've deleted this task:\n%s\nNow you have %s tasks in the list.\n",
+                task, taskTotal);
+        return DIVIDER + s + DIVIDER;
+    }
+
+
+    /**
+     * Shows the message when a task is marked as done.
+     *
+     * @param task Task to be marked.
+     */
+    public String markTask(Task task) {
+        String s = String.format("Nice! I've marked this task as done:\n%s\n", task);
+        return DIVIDER + s + DIVIDER;
+    }
+
+
+    /**
+     * Shows the message when a task is unmarked.
+     *
+     * @param task Task to be unmarked.
+     */
+    public String unmarkTask(Task task) {
+        String s = String.format("OK, I've marked this task as not done yet:\n%s\n", task);
+        return DIVIDER + s + DIVIDER;
+    }
+
+
+    /**
+     * Shows the message when a task is added to the list.
+     *
+     * @param task Task to be added.
+     * @param taskTotal Total number of tasks in the list after adding.
+     */
+    public String addTask(Task task, int taskTotal) {
+        String s = String.format("Got it. I've added this task:\n%s\nNow you have %s tasks in the list.\n",
+                task, taskTotal);
+        return DIVIDER + s + DIVIDER;
+    }
+
+
+    /**
+     * Shows the error message when an exception is thrown
+     *
+     * @param e Exception to be handled.
+     */
+    public String error(DukeException e) {
+        return DIVIDER + e + DIVIDER;
+    }
+
+
+    /**
+     * Shows the message when finding a task.
+     *
+     * @param filteredList Filtered list of tasks.
+     */
+    public String findTask(TaskList filteredList) {
+        if (filteredList.isEmpty()) {
+            return DIVIDER + "There are no matching tasks in your list.\n" + DIVIDER;
+
+        } else {
+            StringBuilder sb = new StringBuilder();
```

Comment:
Perhaps you can initialize the string builder with the leading text, like
`Stringbuilder sb = new StringBuilder("Here are the ...");`